nuxeo-platform-signature-core/src/main/java/org/nuxeo/ecm/platform/signature/core/operations/SignPDF.java

```diff
@@ -54,7 +61,10 @@
     protected DocumentModel doc = null;
 
     @OperationMethod
-    public Blob run(Blob blob) {
+    public Blob run(Blob blob) throws OperationException {
+        if (!((NuxeoPrincipal) ctx.getPrincipal()).isAdministrator()) {
```

Comment:
I would also guard against the case where `ctx.getPrincipal()` is not a `NuxeoPrincipal`, to avoid a `ClassCastException`